test/c/src/cross_serialization_tests.cpp

```diff
@@ -0,0 +1,175 @@
+
+#include <gtest/gtest.h>
+
+#include <cstdbool>
+
+#include "rosidl_runtime_c/message_type_support_struct.h"
+
+#include <rosidl_typesupport_microxrcedds_c/identifier.h>
+#include <rosidl_typesupport_fastrtps_c/identifier.h>
+#include <rosidl_typesupport_introspection_c/identifier.h>
+
+// Workaround for having two coexistent message_type_support_callbacks_t structs
+#define message_type_support_callbacks_t message_type_support_callbacks_fast_t
+#include <rosidl_typesupport_fastrtps_cpp/message_type_support.h>
+#undef message_type_support_callbacks_t
+#define message_type_support_callbacks_t message_type_support_callbacks_xrce_t
+#include <rosidl_typesupport_microxrcedds_c/message_type_support.h>
+#undef message_type_support_callbacks_t
+
+#include "rosidl_typesupport_introspection_c/message_introspection.h"
+#include "rosidl_typesupport_introspection_c/field_types.h"
+
+#include <rosidl_runtime_c/string.h>
+#include <rosidl_runtime_c/string_functions.h>
+
+typedef struct genericSequence {
+  void * data;
+  size_t size;
+  size_t capacity;
+} genericSequence;
+
+template <typename T>
+class CrossSerialization {
+  public:
+
+  #define QUICK_RETURN(x) if(!(x)) return x;
+
+  CrossSerialization(const rosidl_message_type_support_t * type_support_c) :
+    xrce_buffer(nullptr),
+    xrce_size(0),
+    fast_buffer(nullptr),
+    fast_size(0)
+  {
+    xrce_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_microxrcedds_c__identifier);
+    fast_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_fastrtps_c__identifier);
+    introspection_typesupport = get_message_typesupport_handle(type_support_c, rosidl_typesupport_introspection_c__identifier);
+
+    xrce_callbacks = static_cast<const message_type_support_callbacks_xrce_t *>(xrce_typesupport->data);
+    fast_callbacks = static_cast<const message_type_support_callbacks_fast_t *>(fast_typesupport->data);
+    introspection = (rosidl_typesupport_introspection_c__MessageMembers*) introspection_typesupport->data;
+  };
+
+  bool check() {
+    bool ret = true;
+    ret &= xrce_typesupport != nullptr;
+    ret &= fast_typesupport != nullptr;
+    ret &= introspection_typesupport != nullptr;
+    ret &= xrce_callbacks != nullptr;
+    ret &= fast_callbacks != nullptr;
+    ret &= introspection != nullptr;
+    return ret;
+  }
+
+  ~CrossSerialization() {
+    if(nullptr != xrce_buffer) {
+      free(xrce_buffer);
+    }
+    if(nullptr != fast_buffer) {
+      free(fast_buffer);
+    }
+  }
+
+  bool serialize_and_compare_buffers(T & msg) {
+    xrce_size = xrce_callbacks->get_serialized_size(&msg);
+    fast_size = fast_callbacks->get_serialized_size(&msg);
+    QUICK_RETURN(xrce_size == fast_size);
+
+    // size_t xrce_max_size = xrce_callbacks->max_serialized_size();
+    // bool full_bounded;
+    // size_t fast_max_size = fast_callbacks->max_serialized_size(full_bounded);
+    // QUICK_RETURN(xrce_max_size == fast_max_size);
```

Comment:
I am going to take a look into `xrce_callbacks->max_serialized_size();` with bounded arrays.

Reply:
Thanks, go ahead

Reply:
Addresed here: c0132035d91536cc235c8d2baff92b7dc322db9a

`max_serialized_size` generated code for arrays looks like this:
```
  // Member: e
  {
    const size_t array_size = 10;
    for (size_t index = 0; index < array_size; ++index) {
      current_alignment += max_serialized_size_rosidl_typesupport_microxrcedds_test_msg__msg__Regression11(full_bounded, current_alignment);
    }
  }
```

Reply:
Modification reverted to avoid include/link issues